src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentItemDisplayCoordinator.cs

```diff
@@ -96,12 +97,12 @@ public async Task BuildDisplayAsync(ContentItem contentItem, BuildDisplayContext
                     // Create a custom ContentPart shape that will hold the fields for dynamic content part (not implicit parts)
                     // This allows its fields to be grouped and templated 
 
-                    if (part.GetType() == typeof(ContentPart) && contentTypePartDefinition.PartDefinition.Name != contentTypePartDefinition.ContentTypeDefinition.Name)
+                    if (part.GetType() == typeof(ContentPart) && partTypeName != contentTypePartDefinition.ContentTypeDefinition.Name)
                     {
                         var shapeType = context.DisplayType != "Detail" ? "ContentPart_" + context.DisplayType : "ContentPart";
 
                         var shapeResult = new ShapeResult(shapeType, ctx => ctx.ShapeFactory.CreateAsync(shapeType, () => Task.FromResult<IShape>(new ZoneHolding(() => ctx.ShapeFactory.CreateAsync("Zone", Arguments.Empty)))));
-                        shapeResult.Differentiator(contentTypePartDefinition.PartDefinition.Name);
+                        shapeResult.Differentiator(partName);
                         shapeResult.Location("Content");
 
```

Comment:
Looks good @jtkech 
For info @jptissot this `shapeResult.Location("Content");` is why it's also showing up in `SummaryAdmin` - this will make it show everywhere, always. For laters when we know what is the best idea for hiding this, we could do another pr to match whatever is decided.

Reply:
Do you mean we should do a `shapeResult.Location("DetailsAdmin", "-");` too here?

Reply:
I think that's ok here, a dynamic part is composed of fields, so the rendering depends on the field drivers where most of them are not rendered in `SummaryAdmin`.

E.g if a blog post has a dyn part, when displaying the blog in `DetailAdmin`, the related list part will be rendered in `DetailAdmin` but each blogpost will be rendered in `SummaryAdmin`.

The "problem" is when the blog itself has an additional field / part (dynamic or not), it will be rendered when displaying the blog in `DetailAdmin`. But i think that there is nothing to do here, this seems to be only due to the field drivers.

Most of the field drivers already do

    .Location("SummaryAdmin", "")

Maybe we could add

    .Location("DetailAdmin", "")